File: Hotel/hotel.py
def validar_numero_decimal(mensagem):
    while True:
        try:
            numero = float(input(mensagem))
            if numero < 0:
                raise ValueError
            return numero
        except ValueError:
            print("Valor inválido. Por favor, insira um número decimal positivo.")

def validar_numero_inteiro(mensagem):
    while True:
        try:
            numero = int(input(mensagem))
            if numero < 0:
                raise ValueError
            return numero
        except ValueError:
            print("Valor inválido. Por favor, insira um número inteiro positivo.")

def validar_senha():
    senha = input("Por favor, insira sua senha: ")
    return senha == "2678"

def reservar_quarto():
    valor_diaria = validar_numero_decimal("Qual o valor padrão da diária? R$")
    num_diarias = validar_numero_inteiro("Quantas diárias serão necessárias? ")

    nome_hospede = input("Qual o nome do hóspede? ")

    num_quarto = 0
    while num_quarto < 1 or num_quarto > 20:
        num_quarto = validar_numero_inteiro("Qual o quarto para reserva? (1 - 20): ")
        if quartos_ocupados[num_quarto - 1]:
            print("Quarto Ocupado. Escolha outro.")
            num_quarto = 0
        else:
            quartos_ocupados[num_quarto - 1] = True

    valor_total = valor_diaria * num_diarias
    print(f"O valor de {num_diarias} dias de hospedagem é de R${valor_total:.2f}")
    confirmacao = input(f"{nome_hospede}, você confirma a hospedagem por {num_diarias} dias? (S/N): ")
    if confirmacao.upper() == 'S':
        print(f"{nome_hospede}, reserva efetuada. O valor total é de R${valor_total:.2f}.")
    else:
        print("Reserva não efetuada.")

def cadastrar_hospedes():
    valor_diaria = validar_numero_decimal("Qual o valor padrão da diária? R$")

    gratuidades = 0
    meias_hospedagens = 0
    total_custos = 0

    while True:
        nome_hospede = input("Qual o nome do Hóspede? (Digite 'PARE' para encerrar o cadastro) ")
        if nome_hospede.upper() == 'PARE':
            break

        idade_hospede = validar_numero_inteiro("Qual a idade do Hóspede? ")

        custo_diaria = valor_diaria
        if idade_hospede < 6:
            print(f"{nome_hospede} possui gratuidade.")
            gratuidades += 1
        elif idade_hospede > 60:
            custo_diaria /= 2
            print(f"{nome_hospede} paga meia.")
            meias_hospedagens += 1

        total_custos += custo_diaria
        print(f"{nome_hospede} cadastrado(a) com sucesso.")

    print(f"O valor total das hospedagens é: R${total_custos:.2f}; {gratuidades} gratuidade(s); {meias_hospedagens} meia(s)")

def eventos():
    eventos = []
    
    while True:
        print("\n=== Menu de Eventos ===")
        print("1. Adicionar Evento")
        print("2. Listar Eventos")
        print("3. Remover Evento")
        print("4. Sair do Menu de Eventos")
        
        escolha = input("Escolha uma opção: ")
        
        if escolha == "1":
            nome_evento = input("Nome do evento: ")
            data_evento = input("Data do evento (DD/MM/AAAA): ")
            eventos.append({"Nome": nome_evento, "Data": data_evento})
            print("Evento adicionado com sucesso!")
        
        elif escolha == "2":
            print("\n=== Lista de Eventos ===")
            if len(eventos) == 0:
                print("Nenhum evento registrado.")
            else:
                for idx, evento in enumerate(eventos, start=1):
                    print(f"{idx}. {evento['Nome']} - {evento['Data']}")
        
        elif escolha == "3":
            if len(eventos) == 0:
                print("Não há eventos para remover.")
            else:
                print("\n=== Remover Evento ===")
                for idx, evento in enumerate(eventos, start=1):
                    print(f"{idx}. {evento['Nome']} - {evento['Data']}")
                evento_remover = int(input("Informe o número do evento a ser removido: "))
                if evento_remover > 0 and evento_remover <= len(eventos):
                    evento_removido = eventos.pop(evento_remover - 1)
                    print(f"Evento '{evento_removido['Nome']}' removido com sucesso!")
                else:
                    print("Número de evento inválido.")
        
        elif escolha == "4":
            print("Saindo do Menu de Eventos...")
            break
        
        else:
            print("Opção inválida. Tente novamente.")

def abastecimento_carro():
    total_abastecido = 0
    odometro_anterior = 0
    odometro_atual = 0
    consumo_medio = 0
    
    while True:
        print("\n=== Menu de Abastecimento de Carro ===")
        print("1. Registrar Abastecimento")
        print("2. Verificar Consumo Médio")
        print("3. Verificar Total Abastecido")
        print("4. Sair do Menu de Abastecimento")
        
        escolha = input("Escolha uma opção: ")
        
        if escolha == "1":
            litros_abastecidos = float(input("Quantos litros foram abastecidos? "))
            valor_abastecido = float(input("Qual foi o valor total do abastecimento? "))
            odometro_atual = float(input("Informe o odômetro atual do carro: "))
            if odometro_anterior != 0:
                distancia_percorrida = odometro_atual - odometro_anterior
                consumo_atual = litros_abastecidos / distancia_percorrida
                print(f"Consumo atual: {consumo_atual:.2f} km/l")
                consumo_medio = (consumo_medio + consumo_atual) / 2
            total_abastecido += valor_abastecido
            odometro_anterior = odometro_atual
        
        elif escolha == "2":
            if consumo_medio != 0:
                print(f"O consumo médio de combustível é de {consumo_medio:.2f} km/l")
            else:
                print("Nenhum registro de consumo disponível.")
        
        elif escolha == "3":
            if total_abastecido != 0:
                print(f"O total abastecido até agora é de R${total_abastecido:.2f}")
            else:
                print("Nenhum registro de abastecimento disponível.")
        
        elif escolha == "4":
            print("Saindo do Menu de Abastecimento...")
            break
        
        else:
            print("Opção inválida. Tente novamente.")

def manutencao_ar_condicionado():
    while True:
        print("\n=== Menu de Manutenção de Ar-Condicionado ===")
        print("1. Agendar Manutenção")
        print("2. Verificar Status da Manutenção")
        print("3. Cancelar Manutenção")
        print("4. Sair do Menu de Manutenção de Ar-Condicionado")
        
        escolha = input("Escolha uma opção: ")
        
        if escolha == "1":
            data = input("Digite a data para a manutenção (DD/MM/AAAA): ")
            hora = input("Digite o horário para a manutenção (HH:MM): ")
            print(f"Manutenção agendada para {data} às {hora}.")
        
        elif escolha == "2":
            status = input("Digite o código de rastreamento da manutenção: ")
            print(f"O status da manutenção {status} é: Em andamento.")
        
        elif escolha == "3":
            codigo = input("Digite o código de rastreamento da manutenção que deseja cancelar: ")
            confirmacao = input(f"Tem certeza que deseja cancelar a manutenção {codigo}? (S/N): ")
            if confirmacao.upper() == "S":
                print(f"Manutenção {codigo} cancelada com sucesso.")
            else:
                print(f"Cancelamento da manutenção {codigo} abortado.")
        
        elif escolha == "4":
            print("Saindo do Menu de Manutenção de Ar-Condicionado...")
            break
        
        else:
            print("Opção inválida. Tente novamente.")

hospedes = {'João': 101, 'Maria': 102, 'Pedro': 103}

def pesquisar_cadastrar_hospede():
    nome = input("Digite o nome do hóspede para pesquisa ou 'novo' para cadastrar um novo hóspede: ")
    if nome.lower() == 'novo':
        cadastrar_hospede()
    elif nome in hospedes:
        print(f"{nome} está hospedado no quarto {hospedes[nome]}.")
    else:
        print(f"{nome} não está hospedado aqui.")

def cadastrar_hospede():
    nome = input("Digite o nome do novo hóspede: ")
    numero_quarto = input("Digite o número do quarto: ")
    hospedes[nome] = numero_quarto
    print(f"{nome} cadastrado com sucesso no quarto {numero_quarto}.")

def gerenciar_hotel():
    print(f"Bem-vindo ao {nome_hotel}!")
    nome_usuario = input("Qual é o seu nome? ")
    senha_correta = validar_senha()
    if senha_correta:
        print(f"Bem-vindo ao Hotel {nome_hotel}, {nome_usuario}. É um imenso prazer ter você por aqui!")

        while True:
            print("\nMenu de Opções:")
            print("1. Reservar Quarto")
            print("2. Cadastrar Hóspedes")
            print("3. Pesquisar ou Cadastrar Hóspedes")
            print("4. Eventos")
            print("5. Abastecimento de Carro")
            print("6. Manutenção de Ar-Condicionado")
            print("7. Sair")
            opcao = input("Por favor, selecione uma opção: ")

            if opcao == '1':
                reservar_quarto()
            elif opcao == '2':
                cadastrar_hospedes()
            elif opcao == '3':
                pesquisar_cadastrar_hospede()
            elif opcao == '4':
                eventos()
            elif opcao == '5':
                abastecimento_carro()
            elif opcao == '6':
                manutencao_ar_condicionado()
            elif opcao == '7':
                print(f"Muito obrigado e até logo, {nome_usuario}.")
                break
            else:
                print("Opção inválida. Por favor, tente novamente.")

nome_hotel = "Hotel Cleffs"
quartos_ocupados = [False] * 20
gerenciar_hotel()

File: Hotel/hotel.js
var nomeDoHotel = prompt("Qual é o nome do Hotel?");
        alert("O nome do Hotel é " + nomeDoHotel);

        var nomeUsu = prompt("Nos diga seu nome");

        senha()

        inicio();

        function inicio() {

            var escolha = parseInt(prompt(
                'Selecione uma opção: \n' +
                '1. Reserva de Quartos \n' +
                '2. Cadastro de Hóspedes \n' +
                '3. Pesquisa de Hóspedes \n' +
                '4. Abastecimento de Carros \n' +
                '5. Reservar um evento \n' +
                '6. Hora de comer \n' +
                '7. Reserva dos Auditórios \n' +
                '8. Reserva do Restaurante \n' +
                '9. Manutenção do Ar Condicionado \n' +
                '10. Sair'
            ));

            switch (escolha) {
                case 1:
                    resv_quartos();
                    break
                case 2:
                    cadastro_hospede();
                    break
                case 3:
                    cadastroHosp2();
                    break
                case 4:
                    abst_carro();
                    break
                case 5:
                    resv_evento();
                    break
                case 6:
                    comidaParaServir();
                    break
                case 7:
                    auditorio();
                    break
                case 8:
                    resv_restaurante();
                case 9:
                    manutencao_arcondicionado();
                case 10:
                    sair();
                    break
                default:
                    erro();
            }

        }

        function senha() {

            var senhaP = parseInt(prompt("Digite a senha:"));

            if (senhaP === "" || senhaP != 2678) {
                alert("Senha incorreta, tente novamente")
                senha();

            } else {
                alert("Bem vindo ao Hotel " + nomeDoHotel + ", " + nomeUsu + ". é um imenso prazer ter você por aqui!")
                inicio();
            }
        }

        function resv_quartos() {
            var valorDiaria = parseFloat(prompt("Qual o valor padrão da diária?"))
            while (valorDiaria <= 0 || valorDiaria == " ") {
                alert("Valor inválido " + nomeUsu);
                inicio();
            }

            var diaHospedagem = parseInt(prompt("Quantas diárias serão necessárias?"))
            while (diaHospedagem <= 0 || diaHospedagem == " " || diaHospedagem >= 30) {
                alert("Valor inválido " + nomeUsu);
                inicio();
            }

            var somaDiaria = valorDiaria * diaHospedagem

            alert("O valor de " + diaHospedagem + " dias de hospedagem é de R$" + somaDiaria)

            var nomeHospede = prompt("Digite o nome do hóspede:")

            var confirmReserva = prompt(nomeUsu + ", você confirma a hospedagem de " + nomeHospede + " por " + diaHospedagem + " dias? S/N")
            if (confirmReserva == "s") {
                alert(nomeUsu + ", reserva efetuada para " + nomeHospede + ". O valor total é de R$" + somaDiaria)
            } else {
                alert(nomeUsu + ", reseva não efetuada")
            }

            inicio();
        }

        function iniciarReserva() {
            var diaHospedagem = parseInt(prompt("Quantas diárias serão necessárias?"));

            while (isNaN(diaHospedagem) || diaHospedagem <= 0 || diaHospedagem >= 30) {
                alert("Valor inválido. Por favor, insira um número válido de diárias.");
                diaHospedagem = parseInt(prompt("Quantas diárias serão necessárias?"));
            }

            var valorDiaria = 100;
            var somaDiaria = valorDiaria * diaHospedagem;

            alert("O valor de " + diaHospedagem + " dias de hospedagem é de R$" + somaDiaria);

            var nomeHospede = prompt("Digite o nome do hóspede:");

            var confirmReserva = prompt("Você confirma a hospedagem de " + nomeHospede + " por " + diaHospedagem + " dias? (S/N)");
            confirmReserva = confirmReserva.toLowerCase();
            if (confirmReserva === "s") {
                alert("Reserva efetuada para " + nomeHospede + ". O valor total é de R$" + somaDiaria);
            } else {
                alert("Reserva não efetuada.");
            }
        }

        iniciarReserva();

        inicio();

        function cadastro_hospede() {
            var listaHosp = [];
            var gratuidade = 0, meia = 0, inteira = 0;
            var somaMeia = 0, somaInteira = 0;

            var valorPadrao = parseFloat(prompt("Qual o valor padrão da diária?"));
            if (isNaN(valorPadrao) || valorPadrao <= 0) {
                alert('Informe um valor válido para a diária.');
                cadastro_hospede();
                return;
            }

            var nomeHosp = prompt("Informe o nome do hóspede ('pare' para encerrar): ");
            while (nomeHosp.toLowerCase() !== 'pare') {
                var idadeHosp = parseInt(prompt("Qual é a idade do hóspede?"));
                if (isNaN(idadeHosp) || idadeHosp <= 0) {
                    alert('Informe uma idade válida para o hóspede.');
                    nomeHosp = prompt("Informe o nome do hóspede ('pare' para encerrar): ");
                    continue;
                }

                listaHosp.push({ nome: nomeHosp, idade: idadeHosp });

                if (idadeHosp <= 6) {
                    alert(nomeHosp + " cadastrado(a) com sucesso! " + nomeHosp + " possui hospedagem gratuita!");
                    gratuidade++;
                } else if (idadeHosp >= 60) {
                    alert(nomeHosp + ", cadastrado(a) com sucesso! " + nomeHosp + " possui o direito de pagar meia!");
                    meia++;
                    somaMeia += valorPadrao / 2;
                } else {
                    alert(nomeHosp + ", cadastrado(a) com sucesso! " + nomeHosp + " irá pagar inteira!");
                    inteira++;
                    somaInteira += valorPadrao;
                }

                nomeHosp = prompt("Informe o nome do hóspede ('pare' para encerrar): ");
            }

            var total = somaMeia + somaInteira;
            alert("O total de hospedagens é: R$" + total.toFixed(2) + ". " + gratuidade + " hóspede(s) com gratuidade, " + meia + " hóspede(s) com meia e " + inteira + " hóspede(s) pagando inteira.");
            inicio();
        }

        cadastro_hospede();

        function cadastroHosp2() {
            var lista_hospedes = [];

            var escolha_hospedes = parseInt(prompt('Selecione uma opção:\n1.) Cadastrar\n2.) Pesquisar\n3.) Sair'));

            switch (escolha_hospedes) {
                case 1:
                    cadastrar_hospedes();
                    break;
                case 2:
                    pesquisar_hospedes();
                    break;
                case 3:
                    inicio();
                    break;
                default:
                    erroCa2();
            }

            function cadastrar_hospedes() {
                if (lista_hospedes.length >= 15) {
                    alert("Número máximo de hóspedes cadastrados.");
                    cadastroHosp2();
                    return;
                }

                var nome_hospede = prompt('Informe o nome da(o) hóspede:');
                if (!nome_hospede) {
                    alert("Nome inválido.");
                    cadastrar_hospedes();
                    return;
                }

                lista_hospedes.push(nome_hospede);
                alert("Hóspede " + nome_hospede + " foi cadastrado(a) com sucesso!");

                cadastroHosp2();
            }

            function pesquisar_hospedes() {
                var nome_hospede = prompt('Por favor, informe o nome da(o) hóspede para pesquisa:');
                if (lista_hospedes.includes(nome_hospede)) {
                    alert(nome_hospede + ' encontrada(o).')
                } else {
                    alert(nome_hospede + ' não foi encontrada(o).')
                }

                cadastrar_hospedes()
            }

            function erro_pesquisar_hospedes() {
                alert('Por favor, informe um número entre 1 e 3');
                cadastrar_hospedes();
            }
        }
        cadastroHosp2();

        function resv_evento() {
            let duracaoEvento = parseInt(prompt("Qual a duração do evento em horas?"));

            if (isNaN(duracaoEvento) || duracaoEvento <= 0) {
                alert("Por favor, insira uma duração válida para o evento.");
                resv_evento();
                return;
            }

            let quantGar = parseInt(prompt("Quantos garçons serão necessários?"));

            if (isNaN(quantGar) || quantGar <= 0) {
                alert("Por favor, insira um número válido de garçons.");
                resv_evento();
                return;
            }

            let custoTotal = quantGar * 10.50 * duracaoEvento;

            alert("O custo total é de: R$ " + custoTotal.toFixed(2));

            let confirmacao = prompt("Gostaria de efetuar a reserva? (S/N)").toLowerCase();

            if (confirmacao === "s") {
                alert(nomeUsu + ", reserva efetuada com sucesso.");
            }

            inicio();
        }

        function comidaParaServir() {
            let quantConvidados = parseInt(prompt("Qual o número de convidados para o evento?"));

            if (isNaN(quantConvidados) || quantConvidados <= 0) {
                alert("Valor incorreto, digite um número válido maior que zero.");
                comidaParaServir();
                return;
            }

            if (quantConvidados >= 350) {
                alert("A quantidade de convidados é superior à capacidade máxima.");
                inicio();
                return;
            }

            var cafe = 0.20 * quantConvidados;
            var agua = 0.50 * quantConvidados;
            var salgados = 7 * quantConvidados;

            var valorTotal = 0.20 * quantConvidados + 0.40 * quantConvidados + 2.38 * quantConvidados;

            alert("O evento precisará de " + cafe + " litros de café, " + agua + " litros de água e " + salgados + " salgados. O custo do evento será de R$ " + valorTotal.toFixed(2));

            let confirmReserva = prompt("Gostaria de efetuar a reserva? (S/N)").toLowerCase();

            if (confirmReserva === "s") {
                alert(nomeUsu + ", reserva efetuada com sucesso.");
            } else {
                alert("Reserva não efetuada.");
            }

            inicio();
        }

        function auditorio() {
            const capacidadeLaranja = 150;
            const capacidadeColorado = 350;

            let convidados = parseInt(prompt("Qual o número de convidados para seu evento?"));

            while (isNaN(convidados) || convidados < 0 || convidados >= capacidadeColorado) {
                alert("Número de convidados inválido.");
                convidados = parseInt(prompt("Qual o número de convidados para seu evento?"));
            }

            if (convidados < capacidadeLaranja) {
                alert("Use o auditório Laranja.");

                if (convidados > capacidadeLaranja) {
                    let cadeirasAdd = convidados - capacidadeLaranja;
                    alert("Inclua " + cadeirasAdd + " cadeiras adicionais.");
                }
            } else {
                alert("Use o auditório Colorado.");
            }

            let confirmReAud = prompt("Gostaria de efetuar a reserva? (S/N)").toLowerCase();

            if (confirmReAud === "s") {
                alert(nomeUsu + ", reserva efetuada com sucesso.");
            }

            inicio();
        }

        function resv_restaurante() {
            let dia = prompt("Qual é o dia do seu evento?").toLowerCase();

            if (["segunda", "terca", "quarta", "quinta", "sexta"].includes(dia)) {
                let horario = parseInt(prompt("Qual o horário do seu evento?"));

                if (horario < 7 || horario > 23) {
                    alert("Não estamos disponíveis nesse horário.");
                    resv_restaurante();
                    return;
                }

            } else if (dia === "sabado" || dia === "domingo") {
                let horario = parseInt(prompt("Qual é o horário do evento?"));

                if (horario < 7 || horario > 15) {
                    alert("Não estamos disponíveis nesse horário.");
                    resv_restaurante();
                    return;
                }

            } else {
                alert("Dia inválido. Por favor, insira um dia válido.");
                resv_restaurante();
                return;
            }

            var empresa = prompt("Nos informe o nome da empresa: ");
            alert("Restaurante reservado para a empresa " + empresa + " no dia " + dia + " às " + horario + " horas.");

            inicio();
        }

        function abst_carro() {
            const litrosCombustivel = 42;

            let alcoolAzevedo = parseFloat(prompt("Qual o valor do álcool no posto Azevedo Oil?")) * litrosCombustivel;
            let gasolinaAzevedo = parseFloat(prompt("Qual o valor da gasolina no posto Azevedo Oil?")) * litrosCombustivel;
            let alcoolDri = parseFloat(prompt("Qual o valor do álcool no posto Dri Petrol?")) * litrosCombustivel;
            let gasolinaDri = parseFloat(prompt("Qual o valor da gasolina no posto Dri Petrol?")) * litrosCombustivel;

            if (gasolinaAzevedo < gasolinaDri) {
                if (alcoolAzevedo <= (gasolinaAzevedo - (gasolinaAzevedo * 0.3))) {
                    alert(nomeUsu + ", é mais barato abastecer com álcool no posto Azevedo Oil.");
                } else {
                    alert(nomeUsu + ", é mais barato abastecer com gasolina no posto Azevedo Oil.");
                }
            } else {
                if (alcoolDri <= (gasolinaDri - (gasolinaDri * 0.3))) {
                    alert(nomeUsu + ", é mais barato abastecer com álcool no posto Dri Petrol.");
                } else {
                    alert(nomeUsu + ", é mais barato abastecer com gasolina no posto Dri Petrol.");
                }
            }

            inicio();
        }

        function manutencao_arcondicionado() {
            let mais_barata = "";
            let valor_mais_barato = Infinity;

            do {
                let nome_empresa = prompt("Qual o nome da empresa?");
                let valor_aparelho = parseFloat(prompt("Qual o valor do aparelho?"));
                let quantidade_aparelhos = parseInt(prompt("Qual a quantidade de aparelhos?"));
                let desconto = parseInt(prompt("Qual o percentual de desconto?"));
                let minimo_aparelhos = parseInt(prompt("Qual o número mínimo de aparelhos para conseguir o desconto?"));
                let valor_total = valor_aparelho * quantidade_aparelhos;

                if (quantidade_aparelhos >= minimo_aparelhos) {
                    valor_total -= (valor_total * desconto / 100);
                }

                if (valor_total < valor_mais_barato) {
                    mais_barata = nome_empresa;
                    valor_mais_barato = valor_total;
                }

                alert("O serviço de " + nome_empresa + " custará R$" + valor_total.toFixed(2));
                confirmacao_manutencao = prompt("Deseja informar novos dados, " + nomeUsu + "? (S/N)");
            } while (confirmacao_manutencao.toLowerCase() === "s");

            alert("O orçamento de menor valor é da empresa " + mais_barata + " com R$" + valor_mais_barato.toFixed(2));

            inicio();
        }

        function erro() {
            alert('Por favor, informe um número entre 1 e 10.');
            inicio();
        }

        function sair() {
            var confirma = confirm('Você deseja sair?');
            if (confirma) {
                alert("Muito obrigado e até logo, " + nomeUsu + ".");
                window.close();
            } else {
                inicio();
            }
        }

        function erroCa2() {
            alert('Por favor, informe um número entre 1 e 3.');
            cadastroHosp2();
        }